### PR TITLE
support metrics mutex to monitor cost of locks(#26102)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #26102

### CODE REVIEW

File: pkg/util/lock/metric_mutex.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package lock
+
+import (
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/milvus-io/milvus/internal/querycoordv2/params"
```
Comment (jiaoew1991) [inline pkg/util/lock/metric_mutex.go:8]: Don't use the params of this package, just use paramtable.Get().

Comment (MrPresent-Han) [reply pkg/util/lock/metric_mutex.go]: changed

**Pull request merged**